### ISSUE TITLE
Client is in control of the stored file name

### ISSUE THREAD

**Issue by diegonc** [VeliovGroup/Meteor-Files#409]
The client can use the `FSName` option to control the name of the file where the uploaded data is saved by the server. This can be used to... well, evil stuff 😄 

Meteor Files Version: 1.7.13

> **Workaround** Use an onBeforeUpload hook to check the `path` property
> of the uploaded file.
>
> ```javascript
> Collection.onBeforeUpload = function (fileData) {
>     if (/\.\.\//.test(fileData.path)) return "Thou shalt not pass through!" ;
>     return true;
> }
> ```

The client code generating the property is around [line 1700](https://github.com/VeliovGroup/Meteor-Files/blob/master/files.coffee#L1700):
```coffescript
        ...
        @FSName       = if @collection.namingFunction then @collection.namingFunction(@fileData) else @fileId
        ...
```

and the server side uses it without checking around [line 1065](https://github.com/VeliovGroup/Meteor-Files/blob/master/files.coffee#L1065) in `FilesCollection::_prepareUpload`

```coffescript
    ...
    result.path      = "#{@storagePath(result)}#{nodePath.sep}#{opts.FSName}#{extensionWithDot}"
    ...
```

`namingFunction` looks like a feature that should be server-side only. The client can already "set a file name" using the `fileName` property at insert time and should not be granted access to how the file is actually stored as.

BTW, the calls to the naming function made by the server do not supply any parameter. That's odd and doesn't look very useful.

**Comment by dr-dimitru** [comment]
Hello @diegonc ,

Thank you for contribution, could you take a look on https://github.com/VeliovGroup/Meteor-Files/commit/46be3fdc24546ca98c465acc1c17dbb4b2c55244
What do you think?

**Comment by diegonc** [comment]
Yup, looks good.

**Comment by dr-dimitru** [comment]
@diegonc published as [v1.7.14](https://github.com/VeliovGroup/Meteor-Files/releases/tag/1.7.14), thank you for contribution. Let me know how it work for you.

Please, support this project by:
 - Star on [GitHub](https://github.com/VeliovGroup/Meteor-Files)
 - Star on [Atmosphere](https://atmospherejs.com/ostrio/files)
